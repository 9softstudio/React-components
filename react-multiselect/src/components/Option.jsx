import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';

export default class Option extends Component {
    constructor(props) {
        super(props);
    }

    static propTypes = {
        itemData: PropTypes.object.isRequired,
        id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
        onChange: PropTypes.func
    }

    onChange = (event) => {
        event.stopPropagation();
        
        const { itemData, onChange } = this.props;
        
        const newItemData = {
            key: itemData.key,
            value: itemData.value,
            checked: !itemData.checked
        }
        onChange && onChange(newItemData);
    }

    render() {
        const { itemData, id } = this.props;

        return (
            <li className="multiple-select-item">
                <input
                    id={id}
                    type="checkbox"
                    className="option-checkbox"
                    checked={itemData.checked}
                    onChange={this.onChange}
                />
                <label className="option-label" htmlFor={id}>
                    {itemData.value}
                </label>
            </li>
        );
    }
}