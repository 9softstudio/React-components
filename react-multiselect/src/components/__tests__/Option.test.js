import React from 'react'
import renderer from 'react-test-renderer'

import Option from '../Option'

const defaultProps = {
    itemData: {
        key: "key",
        value: "value",
        checked: true
    },
    id: "itemId",
    onChange: jest.fn()
}

describe('Option ', () => {
    afterAll(() => {
        jest.resetAllMocks();
    })

    describe('onChange', () => {
        const event = {
            stopPropagation: jest.fn()
        }

        it('event.stopPropagation should be called', () => {
            const option = new Option(defaultProps);

            option.onChange(event);

            expect(event.stopPropagation).toHaveBeenCalled();
        })

        it('props.onChange should be called with corect parameter', () => {
            const { key, value, checked } = defaultProps.itemData;
            const expectedItemData = { key, value, checked: !checked }

            const option = new Option(defaultProps);

            option.onChange(event);

            expect(defaultProps.onChange).toHaveBeenCalledWith(expectedItemData);
        })
    })

    it('render - content match with snapshot', () => {
        const option = renderer.create(<Option {...defaultProps} />);

        const result = option.toJSON();

        expect(result).toMatchSnapshot();
    })
})