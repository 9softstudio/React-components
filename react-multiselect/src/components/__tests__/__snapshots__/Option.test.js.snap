// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option  render - content match with snapshot 1`] = `
<li
  className="multiple-select-item"
>
  <input
    checked={true}
    className="option-checkbox"
    id="itemId"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="option-label"
    htmlFor="itemId"
  >
    value
  </label>
</li>
`;
